app/jest/my-dashboard/my-teams/teamList.test.tsx

```diff
@@ -0,0 +1,145 @@
+        },
+      ],
+      null,
+    ]),
+  ),
+  deleteTeam: jest.fn(() => Promise.resolve([[''], null])),
+  editTeamName: jest.fn(() => Promise.resolve([, null])),
+}));
+
+describe('Team List', () => {
+  it('Should match the expected button name, and testing on all text-input-box, drop-down-box, hyperlink, and button functionality in the modal', async () => {
+    render(
+      <TeamList
+        currentUser={sampleSession}
+        setTeam={setTeam}
+        loading={false}
+        teams={sampleTeam}
+        loadTeams={loadTeams}
+        hasError={false}
+      />,
+    );
+
+    const createTeamButton = getByRole('button', '+ Create a New Team');
+    expect(createTeamButton).toBeInTheDocument();
+    fireEvent.click(createTeamButton);
+    expect(screen.getByTitle('Create a New Team')).toBeVisible();
+
+    const teamNameInputField = getByLabelText('Team Name');
+    fireEvent.change(await teamNameInputField, { target: { value: 'SAMPLE TEAM 01' } });
+    expect(screen.getByDisplayValue('SAMPLE TEAM 01')).toBeInTheDocument();
+
+    const memberEmailInputField = screen.getByPlaceholderText('Enter email address');
+    fireEvent.change(await memberEmailInputField, { target: { value: 'sampleMember01@gov.bc.ca' } });
+    expect(screen.getByDisplayValue('sampleMember01@gov.bc.ca')).toBeInTheDocument();
+
+    expect(getByRole('option', 'Admin'));
+    expect(getByRole('option', 'Member'));
+    expect(getByRole('link', 'View a detailed breakdown of roles on our wiki page')).toHaveAttribute('href', HYPERLINK);
+
+    const sendInvitationButton = getByRole('button', 'Send Invitation');
+    expect(sendInvitationButton).toBeInTheDocument();
+    await waitFor(() => {
+      fireEvent.click(sendInvitationButton);
+    });
+    expect(createTeam).toHaveBeenCalledTimes(1);
+  });
+
+  it('Should match the correct table headers, selected team', () => {
+    render(
+      <TeamList
+        currentUser={sampleSession}
+        setTeam={setTeam}
+        loading={false}
+        teams={sampleTeam}
+        loadTeams={loadTeams}
+        hasError={false}
+      />,
+    );
+    expect(getByRole('columnheader', 'Team Name'));
+    expect(getByRole('columnheader', 'Actions'));
+    expect(getByRole('row', 'SAMPLE_TEAM Edit Delete')).toHaveClass('active');
+  });
+
+  it('Should be able to click the Delete button, and confirm deletion', async () => {
+    render(
+      <TeamList
+        currentUser={sampleSession}
+        setTeam={setTeam}
+        loading={false}
+        teams={sampleTeam}
+        loadTeams={loadTeams}
+        hasError={false}
+      />,
+    );
+    fireEvent.click(getByRole('button', 'Delete'));
+    expect(screen.getByTitle('Delete team'));
+    await waitFor(() => {
+      fireEvent.click(screen.getByTestId('confirm-delete'));
+    });
+    expect(deleteTeam).toHaveBeenCalledTimes(1);
+  });
+
+  it('Should be able to click the Edit button, and save the new team name', async () => {
+    render(
+      <TeamList
```

Comment:
@kurokn please make sure to remove redundant code

Reply:
put the declarations into a helper.tsx